addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/internal/pca301/Pca301ReadingConverter.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.jeelink.internal.pca301;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.openhab.binding.jeelink.internal.JeeLinkReadingConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Converter for converting a line read from a pcaSerial sketch to a Pca301Reading.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public class Pca301ReadingConverter implements JeeLinkReadingConverter<Pca301Reading> {
+    private static final Pattern READING_P = Pattern.compile(
+            "OK\\s+24\\s+([0-9]+)\\s+4\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+([0-9]+)");
```

Comment:
Pff, these are big and not that maintainable I also see a lot repetition, I would assume it could be resolved smarter with some extra brackets and stating that it should be repeated 8 times if I count correct

```
(\\s+([0-9]+)){8}
```

Reply:
But the problem is of course that you want the captured groups, that sucks... err makes it more difficult...